src/core/mesh/qgsmeshplotter.h

```diff
@@ -0,0 +1,213 @@
+     */
+    void graduatedColors( double value1, double value2, QList<double> &breakValues, QList<QColor> &breakColors, QList<QLinearGradient> &gradients ) const;
+
+  private:
+    QgsColorRampShader mColorRampShader;
+    QColor mSingleColor = Qt::black;
+
+    QgsMeshStrokeColor::ColoringMethod mColoringMethod = SingleColor;
+
+    QLinearGradient makeSimpleLinearGradient( const QColor &color1, const QColor &color2 ) const;
+
+    //! Returns the index of the color ramp shader width value immediatly inferior to value
+    int itemColorIndexInf( double value ) const;
+
+    void graduatedColorsExact( double value1, double value2, QList<double> &breakValues, QList<QColor> &breakColors, QList<QLinearGradient> &gradients ) const;
+    void graduatedColorsInterpolated( double value1, double value2, QList<double> &breakValues, QList<QColor> &breakColors, QList<QLinearGradient> &gradients ) const;
+    void graduatedColorsDiscrete( double value1, double value2, QList<double> &breakValues, QList<QColor> &breakColors, QList<QLinearGradient> &gradients ) const;
+};
+
+/**
+ * \ingroup core
+ *
+ * Class defining stroke width to render dataset. The width can vary depending on dataset value.
+ *
+ * \since QGIS 3.14
+ */
+class CORE_EXPORT QgsMeshStrokeWidth
+{
+  public:
+    //! Returns the minimum value used to defined the varying stroke width
+    double minimumValue() const;
+    //! Sets the minimum value used to defined the varying stroke width
+    void setMinimumValue( double minimumValue );
+
+    //! Returns the maximum value used to defined the varying stroke width
+    double maximumValue() const;
+    //! Sets the maximum value used to defined the varying stroke width
+    void setMaximumValue( double maximumValue );
+
+    //! Returns the minimum width used to defined the varying stroke width
+    double minimumWidth() const;
+    //! Sets the minimum width used to defined the varying stroke width
+    void setMinimumWidth( double minimumWidth );
+
+    //! Returns the maximum width used to defined the varying stroke width
+    double maximumWidth() const;
+    //! Sets the maximum width used to defined the varying stroke width
+    void setMaximumWidth( double maximumWidth );
+
+    //! Returns whether the varying stroke width ignores out of range value
+    bool ignoreOutOfRange() const;
+    //! Sets whether the varying stroke width ignores out of range value
+    void setIgnoreOutOfRange( bool ignoreOutOfRange );
+
+    //! Returns whether the stroke width is varying
+    bool isWidthVarying() const;
+    //! Returns whether the stroke width is varying
+    void setIsWidthVarying( bool isWidthVarying );
+
+    //! Returns the fixed stroke width
+    double fixedStrokeWidth() const;
+    //! Sets the fixed stroke width
+    void setFixedStrokeWidth( double fixedWidth );
+
+    //! Returns the varying stroke width depending on value, if not varying returns the fixed stroke width
+    double strokeWidth( double value ) const;
```

Comment:
I would maybe keep this QgsMeshStrokeWidth class just as a settings container and move this logic mesh edge rendererer